examples/scripts/TreeCompareAuto.C

```diff
@@ -0,0 +1,309 @@
+    // by tree2.StsHit.fX
+    tree1->AddFriend(tree2, "tree2");
+
+    // Define pairs of data members to compare. One from each tree.
+    // This allows to compare them also if names or the structure of
+    // the classses change.
+    std::vector<std::pair<TString, TString>> leaves = GetLeafNames(tree1);
+
+    std::stringstream outstream;
+    Bool_t okay{kTRUE};
+    Int_t numTestedLeaves{0};
+    Int_t numEmptyLeaves{0};
+    Int_t numLeaves{0};
+    Int_t numFailedLeaves{0};
+    Int_t numIdenticalEntries{0};
+    Int_t numIdenticalHistograms{0};
+    Int_t numKolmogorovHistograms{0};
+
+    for (auto leaf : leaves) {
+        TString leafName = leaf.first;
+        TString leafName1 = leaf.second;
+
+        TString command1 = leafName + ">>htemp";
+        tree1->Draw(command1);
+        int entries1{0};
+        float low1{0.};
+        float high1{0.};
+        int nBins1{0};
+        auto htemp = (TH1F*)gPad->GetPrimitive("htemp");
+        if (htemp) {
+            entries1 = htemp->GetEntries();
+            nBins1 = htemp->GetNbinsX();
+            low1 = htemp->GetXaxis()->GetXmin();
+            high1 = htemp->GetXaxis()->GetXmax();
+        }
+
+        command1 = leafName1 + ">>hist1(" + nBins1 + ", " + low1 + ", " + high1 + ")";
+        //    cout << command1 << endl;
+        tree1->Draw(command1);
+        auto hist1 = (TH1F*)gPad->GetPrimitive("hist1");
+        int entries2{0};
+        float low2{0.};
+        float high2{0.};
+        int nBins2{0};
+        if (hist1) {
+            entries2 = hist1->GetEntries();
+            nBins2 = hist1->GetNbinsX();
+            low2 = hist1->GetXaxis()->GetXmin();
+            high2 = hist1->GetXaxis()->GetXmax();
+        }
+
+        if ((0 == entries1 && 0 != entries2) || (0 != entries1 && 0 == entries2)) {
+            std::cout << "One of the distributions is empty" << std::endl;
+            okay = kFALSE;
+        }
+        if (0 == entries1 && 0 == entries2) {
+            outstream << "Both Histograms are empty." << std::endl;
+
+            hist1->Clear();
+            htemp->Clear();
+            //      numTestedLeaves++;
+            //      numEmptyLeaves++;
+            continue;
+        }
+
+        // When executing the draw command "Leaf1 - Leaf2" the subtraction is
+        // executed entry by entry. If the content of the class members are
+        // identical the result is a histogram with a delta function at 0
+        // If the content is differnt one gets a distribution which is
+        // detected.
+        outstream << "Comparing " << leafName << " and " << leafName1 << std::endl;
+        TString command = leafName + "-" + leafName1 + ">>hist(20, -10.,10.)";
+        tree1->Draw(command);
+        auto hist = (TH1F*)gPad->GetPrimitive("hist");
+        numTestedLeaves++;
+
+        // Check if the entries in the tree are identical
+        Bool_t leafIsIdentical = CheckEntriesIdentical(hist, outstream);
```

Comment:
```suggestion

        outstream << "Checking for identical entries" << endl;
        bool leafIsIdentical =    (TMath::Abs(hist->GetMean()) > 0.000001 && TMath::Abs(hist->GetRMS()) > 0.000001)
                               || (0 != hist->GetBinContent(0))
                               || (0 != hist->GetBinContent(hist->GetNbinsX() + 1));
        if (leafIsIdentical) {
            outstream << "Entries are identical." << std::endl;
            outstream << "**********************" << std::endl;
        }
```

* Remove `CheckEntriesIdentical()` -> netto -7 lines of code

Reply:
I would also not re-use the `leafIsIdentical` variable, but have three separate ones:
* `bool const entriesIdentical`
* `bool const histogramsIdentical`
* `bool const kolmogorovIdentical`
or similar. Less mutable variables improve readability for me.

Reply:
fixed

Reply:
I do not remove the small functions since for me they make the code more readable. I only have to look inside if I want to know how the comparison is done.